lux/vislib/altair/Histogram.py

```diff
@@ -45,48 +44,52 @@ def initialize_chart(self):
 
         x_min = self.vis.min_max[msr_attr.attribute][0]
         x_max = self.vis.min_max[msr_attr.attribute][1]
+        x_range = abs(x_max - x_min)
 
         if isinstance(msr_attr.attribute, str):
             msr_attr.attribute = msr_attr.attribute.replace(".", "")
 
         colval = self.vis.data[msr_attr.attribute]
-        x_range = abs(max(colval) - min(colval))
-        plot_range = abs(x_max - x_min)
-        markbar = x_range / plot_range * 12
+        markbar = get_bin_size(self.data[msr_attr.attribute])
+        # Default when IQR was too small
+        if markbar < (x_range / 24):
```

Comment:
So when there are fewer than 24 bins, you use 12? Or am I misreading this?

Reply:
Here, when the `markbar` is so small that there would be *more than* 24 bins, we default to 12. This occurs when the interquartile range is very small compared to the plot range (e.g. most points fall on one value). Thus, it doesn't happen often when plotting the Distribution tab, but moreso when we plot for the Filter tab. This stems from the fact that we use the `x_min` and `x_max` from the original dataset, not the filtered one, when displaying Filter plots (see the second photo for each of the datasets in the Example Test Suite). For example, in the COVID dataset, adding an intent on severity score would yield a graph like the following. If I didn't add the default, the bar would be very skinny, almost invisible. 

![Screen Shot 2021-03-02 at 10 30 16 AM](https://user-images.githubusercontent.com/44735047/109696852-5b9ce900-7b42-11eb-8d2b-97b91cdfab65.png)

Let me know if that clarifies things! As always, I'm open to ideas/improvements. As you see, both the `12` and `24` are a bit arbitrary and worked well after testing on multiple datasets. I wonder if there's a more rigorous approach.

Reply:
Thanks for the explanation. I totally got the comparison inverted in my head.